### ISSUE TITLE
Merge together input data frame with the output of `fit_transform` 

### ISSUE THREAD

**Issue by seaaan** [ropenscilabs/umapr#5]
So you could have the input data frame with two additional columns so the user doesn't need to merge them manually. 

However, user will have to supply only numeric columns to the `umap` function so any categorical labels will be gone. This could possibly be solved by allowing the user to pass in a data frame containing all of their data and then the numeric indices of the columns to subset to. So e.g. you would call `umap(iris, 1:4, ...)` and then we'd run `fit_transform(iris[ , 1:4])` and combine that result with `iris` not `iris[,1:4]`. This seems the most useful to me.
